pandas/core/algorithms.py

```diff
@@ -1334,9 +1334,15 @@ def diff(arr, n: int, axis: AxisInt = 0):
     # added a check on the integer value of period
     # see https://github.com/pandas-dev/pandas/issues/56607
     if not lib.is_integer(n):
-        if not (is_float(n) and n.is_integer()):
```

Comment:
i dont understand how this happens.  n.is_integer should only be called when is_float(n), which should be equivalent to `isinstance(n, (float, np.float_)`.  so the method should always exist